django-stubs/utils/functional.pyi

```diff
@@ -24,7 +24,7 @@ class Promise:
     def __mod__(self, rhs: Any) -> Any: ...
     def __add__(self, other: Any) -> Any: ...
     def __radd__(self, other: Any) -> Any: ...
-    def __deepcopy__(self, memo: Any): ...
+    def __deepcopy__(self: _T, memo: Any) -> _T: ...
```

Comment:
@sobolevn  should we use `typing_extensions.Self` ?

Reply:
Not yet, bound type var is still the way to go.